src/Mvc/Mvc.Core/src/Builder/OpenApiEndpointConventionBuilderExtensions.cs

```diff
@@ -0,0 +1,114 @@
+#pragma warning disable RS0026
+        public static MinimalActionEndpointConventionBuilder Produces(this MinimalActionEndpointConventionBuilder builder,
+#pragma warning restore RS0026
+            int statusCode,
+            Type? responseType = null,
+            string? contentType = null,
+            params string[] additionalContentTypes)
+        {
+            if (responseType is Type && string.IsNullOrEmpty(contentType))
+            {
+                contentType = "application/json";
+            }
+
+            if (contentType is null)
+            {
+                builder.WithMetadata(new ProducesResponseTypeAttribute(responseType ?? typeof(void), statusCode));
+                return builder;
+            }
+
+            builder.WithMetadata(new ProducesResponseTypeAttribute(responseType ?? typeof(void), statusCode, contentType, additionalContentTypes));
+
+            return builder;
+        }
+
+        /// <summary>
+        /// Adds the <see cref="ProducesResponseTypeAttribute"/> with a <see cref="ProblemDetails"/> type
+        /// to <see cref="EndpointBuilder.Metadata"/> for all builders produced by <paramref name="builder"/>. 
+        /// </summary>
+        /// <param name="builder">The <see cref="MinimalActionEndpointConventionBuilder"/>.</param>
+        /// <param name="statusCode">The response status code.</param>
+        /// <param name="contentType">The response content type. Defaults to "application/problem+json".</param>
+        /// <returns>A <see cref="MinimalActionEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+        public static MinimalActionEndpointConventionBuilder ProducesProblem(this MinimalActionEndpointConventionBuilder builder,
+            int statusCode,
+            string contentType = "application/problem+json")
+        {
+            return Produces<ProblemDetails>(builder, statusCode, contentType);
+        }
+
+        /// <summary>
+        /// Adds the <see cref="ProducesResponseTypeAttribute"/> with a <see cref="HttpValidationProblemDetails"/> type
+        /// to <see cref="EndpointBuilder.Metadata"/> for all builders produced by <paramref name="builder"/>. 
+        /// </summary>
+        /// <param name="builder">The <see cref="MinimalActionEndpointConventionBuilder"/>.</param>
+        /// <param name="statusCode">The response status code. Defaults to StatusCodes.Status400BadRequest.</param>
+        /// <param name="contentType">The response content type. Defaults to "application/problem+json".</param>
+        /// <returns>A <see cref="MinimalActionEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+        public static MinimalActionEndpointConventionBuilder ProducesValidationProblem(this MinimalActionEndpointConventionBuilder builder,
+            int statusCode = StatusCodes.Status400BadRequest,
+            string contentType = "application/problem+json")
```

Comment:
Did you say you were going to change this?

Reply:
I had reservations about it because I didn't want to make API changes after the API has been approved.

This is relatively minor and your own on the API review group so I guess it's fine.